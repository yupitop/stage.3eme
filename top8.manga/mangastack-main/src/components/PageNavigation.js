import { Link } from 'react-router-dom';
import { Button } from '@material-ui/core';

const PageNavigation = ({ 
  topOfPage = true, 
  prevLink, 
  nextLink,
  disablePrev,
  disableNext 
}) => (
  <div
    style={{
      display: 'flex',
      alignItems: 'center',
      marginBottom: '3em',
      marginTop: '1em'
    }}>
    <Button
      variant="contained"
      color="primary"
      style={{ flexGrow: 1, marginLeft: 5 }}
      disabled={disablePrev}
      component={Link}
      to={prevLink}
    >
      Prev
    </Button>
    <span style={{ flexGrow: 1 }} />
    <Button
      variant="contained"
      color="primary"
      style={{ flexGrow: 1 }}
      onClick={() => window.scrollTo({
        top: topOfPage ? document.body.scrollHeight : 0,
        left: 0,
        behavior: 'smooth'
      })}
    >
      {topOfPage ? 'Scroll to Bottom' : 'Scroll to Top'}
    </Button>
    <span style={{ flexGrow: 1 }} />
    <Button
      variant="contained"
      color="primary"
      style={{ flexGrow: 1, marginRight: 5 }}
      disabled={disableNext}
      component={Link}
      to={nextLink}
    >
      Next
    </Button>
  </div>
);

export default PageNavigation;