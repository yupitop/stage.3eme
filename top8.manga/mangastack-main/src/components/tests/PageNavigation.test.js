import { MemoryRouter, Route } from 'react-router-dom';
import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import PageNavigation from '../PageNavigation';

let container;
let rerender;
let component;
let testLocation;
beforeEach(() => {
  window.scrollTo = jest.fn();
  component = (page, top = true) => (
    <MemoryRouter initialEntries={['/all?page=2']}>
      <PageNavigation
        topOfPage={top}
        nextLink={`/all?page=${page + 1}`}
        prevLink={`/all?page=${page - 1}`}
        disableNext={page >= 5}
        disablePrev={page <= 1}
      />
      <Route 
        path="*"
        render={({ location }) => {
          testLocation = location;
          return null;
        }}
      />
    </MemoryRouter>
  );
  ({ container, rerender } = render(component(2)));
});

afterEach(cleanup);

it('should render', () => {
  expect(container).toMatchSnapshot();
});

it('should not disable previous and next page buttons when on a middle page', () => {
  const nextButton = screen.getByText('Next');
  const prevButton = screen.getByText('Prev');
  expect(nextButton.closest('a').getAttribute('aria-disabled')).toBe('false');
  expect(prevButton.closest('a').getAttribute('aria-disabled')).toBe('false');
});

it('should navigate to the next page', () => {
  const nextButton = screen.getByText('Next');
  fireEvent.click(nextButton);
  const searchParams = new URLSearchParams(testLocation.search)
  expect(searchParams.has('page')).toBe(true);
  expect(searchParams.get('page')).toBe('3');
});

it('should navigate to the previous page', () => {
  const prevButton = screen.getByText('Prev');
  fireEvent.click(prevButton);
  const searchParams = new URLSearchParams(testLocation.search)
  expect(searchParams.has('page')).toBe(true);
  expect(searchParams.get('page')).toBe('1');
});

it('should disable the next button on the last page', () => {
  rerender(component(5));
  expect(screen.getByText('Next').closest('a').getAttribute('aria-disabled')).toBe('true');
});

it('should disable the previous button on the first page', () => {
  rerender(component(1));
  expect(screen.getByText('Prev').closest('a').getAttribute('aria-disabled')).toBe('true');
});

it('should display a scroll to bottom button if it is at the top of the page', () => {
  const scrollToBottomButton = screen.getByText('Scroll to Bottom');
  expect(scrollToBottomButton).toBeInTheDocument();
  fireEvent.click(scrollToBottomButton);
  expect(window.scrollTo).toHaveBeenLastCalledWith({ behavior: 'smooth', left: 0, top: 0 });
});

it('should display a scroll to top button if it is at the bottom of the page', () => {
  rerender(component(2, false));
  const scrollToTopButton = screen.getByText('Scroll to Top');
  expect(scrollToTopButton).toBeInTheDocument();
  fireEvent.click(scrollToTopButton);
  expect(window.scrollTo).toHaveBeenLastCalledWith({ behavior: 'smooth', left: 0, top: 0 });
});
