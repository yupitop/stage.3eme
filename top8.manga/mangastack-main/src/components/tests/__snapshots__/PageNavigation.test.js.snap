// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <div
    style="display: flex; align-items: center; margin-bottom: 3em; margin-top: 1em;"
  >
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      href="/all?page=1"
      role="button"
      style="flex-grow: 1; margin-left: 5px;"
      tabindex="0"
    >
      <span
        class="MuiButton-label"
      >
        Prev
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
    <span
      style="flex-grow: 1;"
    />
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      style="flex-grow: 1;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Scroll to Bottom
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <span
      style="flex-grow: 1;"
    />
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      href="/all?page=3"
      role="button"
      style="flex-grow: 1; margin-right: 5px;"
      tabindex="0"
    >
      <span
        class="MuiButton-label"
      >
        Next
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
  </div>
</div>
`;
